src/engine/controls/cuecontrol.cpp

```diff
@@ -139,6 +141,76 @@ CueControl::~CueControl() {
     qDeleteAll(m_hotcueControls);
 }
 
+mixxx::audio::FramePos CueControl::nextTrigger(bool reverse,
+        mixxx::audio::FramePos currentPosition,
+        mixxx::audio::FramePos* pTargetPosition,
+        mixxx::audio::FrameDiff_t lookAheadFrames) {
+    VERIFY_OR_DEBUG_ASSERT(pTargetPosition) {
+        return mixxx::audio::kInvalidFramePos;
+    }
+    *pTargetPosition = mixxx::audio::kInvalidFramePos;
+    mixxx::audio::FramePos triggerPosition = mixxx::audio::kInvalidFramePos;
+    HotcueControl* pNextJump = nullptr;
+    for (const auto& pControl : std::as_const(m_hotcueControls)) {
```

Comment:
A brief description as source code comment would be nice what this loop is doing.

Reply:
Added a comment but not sure if it is very descriptive. Feel free to suggest a change on it!